loader/src/ui/nodes/TextAreaV2.cpp

```diff
@@ -0,0 +1,377 @@
+}
+
+void geode::TextAreaV2::resetMaxBounds() {
+    this->resetMaxWidth();
+    this->resetMaxHeight();
+}
+
+void geode::TextAreaV2::setDeferUpdates(const bool defer) {
+    m_deferUpdates = defer;
+
+    // If already deferred, update the container immediately and unschedule the deferred update
+    if (m_deferred) {
+        this->unschedule(schedule_selector(TextAreaV2::updateContainer));
+        
+        this->updateContainer();
+    }
+}
+
+float geode::TextAreaV2::getLineHeight() const {
+    return m_lines.empty() ? 0 : m_lines.front().label->getScaledContentHeight() + m_linePadding;
+}
+
+void geode::TextAreaV2::update() {
+    if (!m_deferred) {
+        if (m_deferUpdates) {
+            m_deferred = true;
+
+            this->scheduleOnce(schedule_selector(TextAreaV2::updateContainer), 0);
+        } else {
+            this->updateContainer();
+        }
+    }
+}
+
+void geode::TextAreaV2::updatePadding() {
+    this->update();
+}
+
+void geode::TextAreaV2::updateContainer(const float dt) {
+    PaddingNode::updatePadding();
+    m_container->removeAllChildren();
+
+    m_deferred = false;
+    m_lines = this->createLines();
+
+    const bool upperBoundWidth = m_maxWidth >= 0;
+    const bool upperBoundHeight = m_maxHeight >= 0;
+    const bool artificialWidth = m_minWidth >= 0 || upperBoundWidth;
+    const bool artificialHeight = m_minHeight >= 0 || upperBoundHeight;
+    const bool inheritedNodeSize = this->getContentSize() == this->getPaddedContainerSize();
+    float height = m_lines.empty() ? 0 : m_lines.back().currentHeight;
+    float width = 0;
+
+    // First determine the container size before manipulating the node anchors and true positions
+    for (const Line& line : m_lines) {
+        width = std::max(width, line.label->getScaledContentWidth());
+    }
+
+    if (artificialWidth) {
+        width = std::max(m_minWidth, width);
+
+        m_container->setContentWidth(upperBoundWidth ? std::min(m_maxWidth, width) : width);
+    } else {
+        m_container->setContentWidth(width);
+    }
+
+    if (artificialHeight) {
+        height = std::max(m_minHeight, height);
+
+        m_container->setContentHeight(upperBoundHeight ? std::min(m_maxHeight, height) : height);
+    } else {
+        m_container->setContentHeight(height);
+    }
+
+    if (inheritedNodeSize) {
+        this->setContentSize(this->getPaddedContainerSize());
+    }
+
+    for (const Line& line : m_lines) {
+        this->updateLineAlignment(line.label);
+
+        // Correct the Y position to be relative to the container height
+        line.label->setPositionY(m_container->getContentHeight() - line.currentHeight + line.label->getScaledContentHeight());
+        m_container->addChild(line.label);
+    }
+}
+
+void geode::TextAreaV2::updateLineAlignment(CCLabelBMFont* line) {
+    switch (m_alignment) {
+        case Alignment::Left:
+            line->setAnchorPoint({ 0, 1 });
+            line->setPositionX(0);
+            break;
```

Comment:
the `break` along with the brace mentioned above should be on the same indent as the `case` for legibility and there should be a newline afterwards